src/core/agent/services/connectionService.ts

```diff
@@ -360,29 +357,26 @@ class ConnectionService extends AgentService {
         await this.createConnectionMetadata(contact.id, {
           alias: contact.alias,
           oobi: contact.oobi,
+          groupId: contact.groupCreationId,
+          createdAtUTC: contact.createdAt,
         });
       }
     }
   }
 
   @OnlineOnly
-  async resolveOobi(url: string, waitForCompletion = true): Promise<any> {
-    const startTime = Date.now();
+  async resolveOobi(
+    url: string,
+    waitForCompletion = true
+  ): Promise<Operation & { response: State } & Contact> {
     const alias = new URL(url).searchParams.get("name") ?? uuidv4();
-    let operation;
-    if (waitForCompletion) {
-      operation = await waitAndGetDoneOp(
-        this.props.signifyClient,
-        await this.props.signifyClient.oobis().resolve(url, alias),
-        5000
-      );
-    } else {
-      operation = await waitAndGetDoneOp(
-        this.props.signifyClient,
-        await this.props.signifyClient.oobis().resolve(url, alias),
-        2000 - (Date.now() - startTime)
-      );
-    }
+    const groupId = new URL(url).searchParams.get("groupId") ?? "";
+    const operation = (await waitAndGetDoneOp(
+      this.props.signifyClient,
+      await this.props.signifyClient.oobis().resolve(url, alias),
```

Comment:
What happens if not `alias`?
We should throw an error before

Reply:
We are changing the UI so that if there's no `alias` in the OOBI URL, the user can enter their own name for the connection before we start resolving the OOBI.